component/client/settings/ClientSetting.tsx

```diff
@@ -93,7 +94,8 @@ class ClientSetting extends React.Component<any, IClientSettingState> {
     event.preventDefault();
     if (event.target.value !== undefined) {
       this.setState({
-        dirty: this.state.username.original !== event.target.value,
+        // The field is only considered dirty if there is value and it's different from original
+        dirty: event.target.value && this.state.username.original !== event.target.value,
         username: {
           original: this.state.username.original,
```

Comment:
One thing to note is if you want to update state based on the previous state, you should never access previous it directly.
You should use a callback -- this.setState((prevState, props) => {  return newState  });
this.state may not be updated as Rect specified. 

Refer to https://facebook.github.io/react/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous